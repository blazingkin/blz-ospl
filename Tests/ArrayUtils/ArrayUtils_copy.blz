:main
    #Test 00 :copy
	print(">>> Test 00 :copy <<<")
	arr1 = [0,1,2]
	arr2 = arr1.copy()
	arr2[0] = 4
	print("arr1: " + arr1)
	print("arr2: " + arr2)
	print("--------------------------------")
	
	#Test 01 :copy!
	print(">>> Test 00 :copy! <<<")
	arr3 = arr1.copy!()
	print("arr3: " + arr3)
	arr3[0] = {nil}
	print("arr3: " + arr3)
	print("arr1: " + arr1)
	print("--------------------------------")
end