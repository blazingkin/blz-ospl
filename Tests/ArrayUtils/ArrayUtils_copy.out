>>> Test 00 :copy <<<
arr1: [0, 1, 2]
arr2: [4, 1, 2]
--------------------------------
>>> Test 00 :copy! <<<
arr3: [0, 1, 2]
arr3: [, 1, 2]
arr1: [, 1, 2]
--------------------------------
